### ISSUE TITLE
Confirm that returning 2 fids when asked for 3 does not generate MATLAB exception

### ISSUE THREAD

**Issue by jeanconn** [sot/proseco#150]
Looks like the current fid optimization code returns 2 fids even if 3 are requested for the undercover (I'm looking at obsid 14277).  I thought it would return 3 even if one was off the CCD, so didn't explicitly handle this.

**Comment by jeanconn** [comment]
It looks like despite the fact I didn't quite understand our API, that this doesn't crash things at the star selection stage.  I didn't immediately find a modern (ORVDOT) workspace with a fid off the CCD to test for no crash exporting a DOT from ORViewer.

**Comment by taldcroft** [comment]
Sorry if that wasn't clear, but I put a fair bit of effort into only returning a catalog of usable fids.  Seems like the right answer, previous behavior was basically broken, and hopefully this is the time to fix it.

**Comment by taldcroft** [comment]
ACIS undercover is literally the first item on our test requirements wiki, so this won't be forgotten.

**Comment by jeanconn** [comment]
I completely understand the desire to just "do this right", but I am concerned that returning 2 fids when asked for 3 doesn't win us anything operationally (we still end up with 2 tracked fids and 5 guide stars, not 2 tracked fids and 6 guide stars) and somewhat breaks consistency on the MATLAB side.  

On the other hand, my concern was the mismatch (ask 3 get and assign 2) would cause an unhandled exception on assignment or editing, but that hasn't happened in my testing thus far.  And I'm not concerned about this not getting tested, I was just concerned about timelines and rework if needed.

**Comment by taldcroft** [comment]
From the proseco side, changing this now will definitely be substantial rework.  It just gets awkward all around.  The relevant code for rejecting candidates is:
```
        for idx, fid in enumerate(cand_fids):
            if (self.off_ccd(fid) or
                    self.near_hot_or_bad_pixel(fid) or
                    self.has_column_spoiler(fid, self.stars, stars_mask)):
                idx_bads.append(idx)
```
Being off the CCD entirely is actually worse than the other two, so the simple implementation would be to entirely ignore all these tests if it would reduce the number of candidates below `n_fid`.

So for example imagine a large sim_offset that puts ACIS 4-6 off the CCD, and ACIS-1 has a column spoiler.  If processing in order, fid 1 would get rejected, and then 4 and 5, but finally the off-CCD fid 6 would get accepted to ensure that there are at least 3 candidates.  Granted, this is almost inconceivable, but still it is annoying to easy see corner cases that give the wrong behavior.

Anyway, your concerns are well-founded.  So let's just stay aware of this, add test cases, and not change proseco unless it is driven by Matlab test failures that cannot be easily fixed.

**Comment by jeanconn** [comment]
On the proseco side, it seems like a small code block to check if returning fewer than the asked-for fids and add any crap fid from the not-picked set if needed.  But agreed to just test.

**Comment by taldcroft** [comment]
We don't need to track this as a separate issue since it is included in the list of tests in #188 that the FOT are actually looking at.
